Fix `Cargo.toml`

jyhi opened the pull request:
# Description of change

This PR tries to improve the quality of `Cargo.toml`s.

## Links to any relevant issues

Fix #798.

## Type of change

- Chore

## How the change has been tested

```
cargo test --all-features
```

Make sure to provide instructions for the maintainer as well as any relevant configurations.

## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that new and existing unit tests pass locally with my changes

jyhi committed:
**/Cargo.toml: change authors to IOTA Stiftung

Fixes #798.

jyhi committed:
**/Cargo.toml: upgrade to Rust 2021

jyhi committed:
**/Cargo.toml: fix metadata

Unify names, descriptions, and licensing information.

jyhi committed:
**/Cargo.toml: sort dependencies

- Add `default-features = false` for all dependencies
- Sort dependency names alphabetically

jyhi committed:
bindings/java: fix typo in Cargo.toml description

Thoralf-M commented on bindings/nodejs/Cargo.toml:
```diff
@@ -1,32 +1,35 @@
 [package]
-name = "wallet"
+name = "iota-wallet-nodejs"
```
We can't just change it because of `The crate name will be read from the npm_package_name environment variable. If the package name includes a namespace (@namespace/package), the namespace will be removed when matching the crate name (package).` and our npm package is `@iota/wallet`
https://github.com/neon-bindings/cargo-cp-artifact

jyhi committed:
bindings/nodejs: fix crate name

cargo-cp-artifact reads the crate name and matches it with the one
specified in `package.json`, so they must be the same.

rajivshah3 commented on Cargo.toml:
```diff
@@ -1,8 +1,13 @@
 version = "0.2.0"
-authors = ["Lucas Nogueira <lucas.nogueira@iota.org>"]
-edition = "2018"
+authors = ["IOTA Stiftung"]
```
For the author email should we put `contact@iota.org`?

jyhi replied:
The change is borrowed from [bee](https://github.com/iotaledger/bee/blob/dev/bee-common/bee-common/Cargo.toml#L4); perhaps let them know this concern too.

jyhi committed:
**/Cargo.toml: add entries

- homepage
- categories
